flocker/control/_config.py

```diff
@@ -1257,7 +1262,10 @@ def deployment_from_configuration(deployment_configuration, all_applications):
                         hostname=hostname, application_name=name)
                 )
             node_applications.append(application)
-        node = Node(hostname=hostname,
+        if hostname not in node_states:
+            raise ConfigurationError("No known node with address {}.".format(
+                hostname))
```

Comment:
Ah, I see. That's nice.
So this is somewhat related to https://clusterhq.atlassian.net/browse/FLOC-1278 `Check that the primary node a dataset is created on is actually part of the cluster`